bpf/proxy/lb_src_range_test.go

```diff
@@ -36,193 +36,212 @@ func init() {
+					net.IPv4(10, 0, 0, 2),
+					2222,
+					v1.ProtocolTCP,
+					proxy.K8sSvcWithExternalIPs(extIPs),
+					proxy.K8sSvcWithLoadBalancerIPs(lbIPs),
+					proxy.K8sSvcWithLBSourceRangeIPs([]string{"35.0.1.2/24"}),
+				)
+
+				err := s.Apply(state)
+				Expect(err).NotTo(HaveOccurred())
+				Expect(svcs.m).To(HaveLen(3))
+
+				Expect(svcs.m).To(HaveKey(keyWithSaddr1))
+				Expect(svcs.m).NotTo(HaveKey(keyWithSaddr3))
+
+			}))
+
+			By("Deleting LBSourceRangeIP for existing service", makestep(func() {
+				state.SvcMap[svcKey] = proxy.NewK8sServicePort(
+					net.IPv4(10, 0, 0, 2),
+					2222,
+					v1.ProtocolTCP,
+					proxy.K8sSvcWithExternalIPs(extIPs),
+					proxy.K8sSvcWithLoadBalancerIPs(lbIPs),
+					proxy.K8sSvcWithLBSourceRangeIPs([]string{}),
+				)
+
+				err := s.Apply(state)
+				Expect(err).NotTo(HaveOccurred())
+				Expect(svcs.m).To(HaveLen(2))
+
+				Expect(svcs.m).NotTo(HaveKey(keyWithSaddr1))
+				Expect(svcs.m).NotTo(HaveKey(keyWithSaddr3))
+
+			}))
+
+			By("Adding new entries to the map with different source IPs", makestep(func() {
+				var tempLbIPs, tempExtIPs []string
+				if len(extIPs) == 0 {
+					tempLbIPs = append(lbIPs, "45.0.1.2")
+				} else {
+					tempExtIPs = append(extIPs, "45.0.1.2")
+				}
+				state.SvcMap[svcKey] = proxy.NewK8sServicePort(
+					net.IPv4(10, 0, 0, 2),
+					2222,
+					v1.ProtocolTCP,
+					proxy.K8sSvcWithExternalIPs(tempExtIPs),
+					proxy.K8sSvcWithLoadBalancerIPs(tempLbIPs),
+					proxy.K8sSvcWithLBSourceRangeIPs([]string{"33.0.1.2/24", "38.0.1.2/16", "40.0.1.2/32"}),
+				)
+
+				err := s.Apply(state)
+				Expect(err).NotTo(HaveOccurred())
+				Expect(svcs.m).To(HaveLen(9))
+				s.Stop()
+			}))
+
+			By("Remove stale src range entries after syncer restarts", makestep(func() {
+				state.SvcMap[svcKey] = proxy.NewK8sServicePort(
+					net.IPv4(10, 0, 0, 2),
+					2222,
+					v1.ProtocolTCP,
+					proxy.K8sSvcWithExternalIPs(extIPs),
+					proxy.K8sSvcWithLoadBalancerIPs(lbIPs),
+					proxy.K8sSvcWithLBSourceRangeIPs([]string{"35.0.1.2/24"}),
+				)
+				s, _ = proxy.NewSyncer(nodeIPs, svcs, eps, aff, rt)
+				err := s.Apply(state)
+				Expect(err).NotTo(HaveOccurred())
+				Expect(svcs.m).To(HaveLen(3))
+			}))
+
+			By("Remove all stale src ranges after syncer restarts", makestep(func() {
+				state.SvcMap[svcKey] = proxy.NewK8sServicePort(
+					net.IPv4(10, 0, 0, 2),
+					2222,
+					v1.ProtocolTCP,
+					proxy.K8sSvcWithExternalIPs(extIPs),
+					proxy.K8sSvcWithLoadBalancerIPs(lbIPs),
+				)
+				s, _ = proxy.NewSyncer(nodeIPs, svcs, eps, aff, rt)
+				err := s.Apply(state)
+				Expect(err).NotTo(HaveOccurred())
+				Expect(svcs.m).To(HaveLen(2))
+			}))
+
+			By("deleting the services", makestep(func() {
+				delete(state.SvcMap, svcKey)
+
+				err := s.Apply(state)
+				Expect(err).NotTo(HaveOccurred())
+
+				Expect(svcs.m).To(HaveLen(0))
+				Expect(eps.m).To(HaveLen(0))
+			}))
+
+		})
+	}
+	testfn([]string{}, []string{"35.0.0.2"})
```

Comment:
Since the two test cases are pretty much mutually exclusive, you could just pass the one option.